src/libs/elektra/kdb.c

```diff
@@ -1006,6 +1007,14 @@ KDB * kdbOpen (const KeySet * contract, Key * errorKey)
 		goto error;
 	}
 
+	// TOOD (atmaxinger): improve
+	if (initHooks (handle, ksDup (elektraKs), handle->modules, errorKey) == -1)
+	{
+		ELEKTRA_SET_INSTALLATION_ERROR (errorKey, "Mounting hooks failed. Please see warning of concrete plugin");
+		ksDel (elektraKs);
+		goto error;
+	}
```

Comment:
You should probably use the same trick as in `runSetPhase` to make sure the plugin can only create warnings (*), otherwise this `ELEKTRA_SET_INSTALLATION_ERROR` may actually result in a warning.

https://github.com/ElektraInitiative/libelektra/blob/6dc6bb8305ff8d9581f2ffa518d9120045f21eae/src/libs/elektra/kdb.c#L2058-L2063

https://github.com/ElektraInitiative/libelektra/blob/6dc6bb8305ff8d9581f2ffa518d9120045f21eae/src/libs/elektra/kdb.c#L2158-L2164

You may need to modify the `if` conditions a bit. The idea is simply that setting any value for `meta:/error` will block the creation of errors.

---

(*) In case you don't know, the function behind the `ELEKTRA_SET_*_ERROR` macros will create a warning instead of an error, if there is already an error. That's because according to our design, a `Key` may only contain a single error.